conans/test/unittests/tools/env/test_env.py

```diff
@@ -0,0 +1,284 @@
+import os
+import platform
+import subprocess
+import textwrap
+
+import pytest
+
+from conan.tools.env import Environment
+from conan.tools.env.environment import ProfileEnvironment
+from conans.client.tools import chdir
+from conans.test.utils.test_files import temp_folder
+from conans.util.files import save
+
+
+def test_compose():
+    env = Environment()
+    env.define("MyVar", "MyValue")
+    env.define("MyVar2", "MyValue2")
+    env.define("MyVar3", "MyValue3")
+    env.define("MyVar4", "MyValue4")
+    env.unset("MyVar5")
+
+    env2 = Environment()
+    env2.define("MyVar", "MyNewValue")
+    env2.append("MyVar2", "MyNewValue2")
+    env2.prepend("MyVar3", "MyNewValue3")
+    env2.unset("MyVar4")
+    env2.define("MyVar5", "MyNewValue5")
+
+    env.compose(env2)
+    assert env.value("MyVar") == "MyNewValue"
+    assert env.value("MyVar2") == 'MyValue2 MyNewValue2'
+    assert env.value("MyVar3") == 'MyNewValue3 MyValue3'
+    assert env.value("MyVar4") == ""
```

Comment:
The operation done at the system environment will be an unset. This is a unittest checking the internal representation of the ``Environment``, which reads more like a set of actions to do on the environment rather than actual environment variable values.